bin/agent-data-plane/Cargo.toml

```diff
@@ -10,6 +10,7 @@ default = []
 fips = ["saluki-app/tls-fips"]
 
 [dependencies]
+arc-swap = { workspace = true }
```

Comment:
Shouldn't need to add this here since you're not actually using it in the binary crate.